src/reducer/cart-reducer.js

```diff
@@ -0,0 +1,41 @@
+export const cartReducer = (cartState, action) => {
+
+    switch(action.type) {
+
+        case "ADD-TO-CART":
+            return {
+                ...cartState,
+                ...cartState.product,
+                ...cartState.adder,
+                adder: !cartState.adder,
+                product: action.payload
+            }
+        
+        case "DELETE-FROM-CART":
+            return {
+                ...cartState,
+                ...cartState.productId,
+                ...cartState.deleteProduct,
+                ...cartState.deleter,
+                deleter: !cartState.deleter,
+                deleteProduct: true,
+                productId: action.payload
+            }
```

Comment:
I am not able to understand the logic here could you explain a bit what we're trying to do